src/custom/components/swap/EthWethWrap/index.tsx

```diff
@@ -79,13 +85,31 @@ export interface Props {
   account?: string
   native: Currency
   wrapped: Token
+  wrapCallback: () => Promise<void>
 }
 
-export default function EthWethWrap({ account, native, wrapped }: Props) {
+export default function EthWethWrap({ account, native, wrapped, wrapCallback }: Props) {
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState<Error | null>(null)
+  const [nativeBalance, wrappedBalance] = useCurrencyBalances(account, [native, wrapped])
+
   const wrappedSymbol = wrapped.symbol || 'wrapped native token'
   const nativeSymbol = native.symbol || 'native token'
 
-  const [nativeBalance, wrappedBalance] = useCurrencyBalances(account, [native, wrapped])
+  const handleWrap = useCallback(async () => {
+    try {
+      setLoading(true)
+      setError(null)
+
+      await wrapCallback()
+    } catch (error) {
+      console.error('Error wrapping ETH:', error)
+
+      setError(new Error(error))
+    } finally {
+      setLoading(false)
```

Comment:
I see how this is simpler than the original approach we discussed. I think we can do it in a separate PR the part where if there's at least one WRAPING going on, we are "loading".

Reply:
don't understand this comment but let's discuss everything in another PR, merging this to consolidate